src/Chapter1.hs

```diff
@@ -541,8 +541,14 @@ value after "=" where the condition is true.
 
 Casual reminder about adding top-level type signatures for all functions :)
 -}
-
-mid x y z = error "mid: not implemented!"
+mid :: Ord a => a -> a -> a -> a
+mid x y z
+  | x <= y && y <= z = y
+  | x <= z && z <= y = z
+  | y <= x && x <= z = x
+  | y <= z && z <= x = z
+  | z <= x && x <= y = x
+  | otherwise = y
```

Comment:
You can actually get rid of the first case as well, to return `y` on those two cases at the same time and have fewer lines in your implementation :slightly_smiling_face: